plugins/pulp_rpm/plugins/controllers/errata.py

```diff
@@ -0,0 +1,44 @@
+import mongoengine
+
+from pulp_rpm.plugins.db.models import Errata, ErratumPkglist
+
+
+def create_or_update_pkglist(erratum, repo_id):
+    """
+    Create or update erratum pkglist and save it to DB.
+    Also make pkglist on Errata instance empty (in memory, not saved to DB yet).
+
+    :param erratum: An erratum unit being imported
+    :type erratum: pulp_rpm.plugins.db.models.Errata
+    :param repo_id: An id of repository into which the erratum unit is being imported
+    :type repo_id: str
+    """
+    pkglist_info = {'errata_id': erratum.errata_id,
+                    'repo_id': repo_id,
+                    'collections': erratum.pkglist}
+    new_pkglist = ErratumPkglist(**pkglist_info)
+    try:
+        new_pkglist.save()
+    except mongoengine.NotUniqueError:
+        # check if erratum unit exists or should be updated, if so - update its pkglist
+        existing_erratum = Errata.objects.filter(**erratum.unit_key).first()
+        new_erratum = erratum
+        if not existing_erratum or existing_erratum.update_needed(new_erratum):
+            existing_pkglist = ErratumPkglist.objects.filter(**new_pkglist.model_key).first()
+            existing_pkglist.collections = new_pkglist.collections
+            existing_pkglist.save()
+    # pkglist is saved to a separate collection, we no longer need it in the erratum unit itself
+    erratum.pkglist = []
+
+
+def get_pkglists(errata_id):
+    """
+    Get erratum pkglists for a given errata_id.
+
+    :param errata_id: An id of erratum unit
+    :type erratum: str
+
+    :return: erratum pkglists
+    :rtype: mongoengine.queryset.QuerySet
+    """
+    return ErratumPkglist.objects(errata_id=errata_id)
```

Comment:
I love the function above. I think the whole pattern is more natural now. But this one I'm less crazy about.

This only operates on a single collection; it happens to be searching based on a foreign key, but it's not interacting with multiple collections. I don't see that 

``get_pkglists('foo')``

is better than just writing

``ErratumPkglist.objects(errata_id='foo')``

What do you think about getting rid of this function and replacing it with the django API usage in the two places where it gets used?